tools/ncf.py

```diff
@@ -221,8 +221,23 @@ def parse_technique_methods(technique_file):
             promise_class_context = class_context_and(class_context, attribute['rval']['value'])
           if attribute['rval']['type'] == 'functionCall':
             ifvarclass_function = attribute['rval']['name']
-            ifvarclass_args = [arg['value'].replace('\\"', '"') for arg in attribute['rval']['arguments']]
-            promise_class_context = class_context_and(class_context, ifvarclass_function + '(' + ','.join(ifvarclass_args) + ')')
+            if ifvarclass_function == 'concat':
+              ifvarclass_args = []
+              for arg in attribute['rval']['arguments']:
+                print ifvarclass_args
+                if arg['type'] == 'string':
+                  ifvarclass_args.append(arg['value'])
+                elif arg['type'] == 'functionCall' and arg['name'] == 'canonify':
+                  ifvarclass_args.append(arg['arguments'][0]['value'])
+                else:
+                    function_name = arg['name']
+                    function_args = [function_arg['value'].replace('\\"', '"') for function_arg in arg['arguments']]
+                    joined_args = fvarclass_function + '(' + ','.join(ifvarclass_args) + ')'
+                    ifvarclass_args.append(joined_args)
+              promise_class_context = ''.join(ifvarclass_args)
+            else:
+              ifvarclass_args = [arg['value'].replace('\\"', '"') for arg in attribute['rval']['arguments']]
+              promise_class_context = class_context_and(class_context, ifvarclass_function + '(' + ','.join(ifvarclass_args) + ')')
```

Comment:
Can you tell me if I right : this patch removes canonify that are within concat calls ?